### PR TITLE
TIG-1579 Always remove toolchain before compilation

### PR DESCRIPTION
**Pull request by guoyr**

### CODE REVIEW

File: evergreen.yml [hunk 1]
```diff
@@ -362,6 +349,11 @@ functions:
       script: |
         set -eo pipefail
 
+        # Hosts may have incomplete or invalid toolchain directories from
```
Comment (visemet) [inline evergreen.yml]: "3. Put comments in yaml not in inline shell."

File: evergreen.yml [hunk 2]
```diff
@@ -119,10 +119,6 @@ tasks:
       short_name: genny
       install_dir: dist
 
-- name: t_rm_toolchain_dir_static_host
```
Comment (visemet) [inline evergreen.yml]: Seems like https://github.com/mongodb/genny/blob/336a7cca9525b066a3425d7775381668f6fb11e3/evergreen.yml#L221 needs to also be removed. It's probably worth keeping the `tg_compile_static_host` task group for consistency and ease of adding new tasks back.

Comment (guoyr) [reply evergreen.yml]: Yep.

I also changed all hosts to run compile in a task group to be consistent with other hosts

File: evergreen.yml (hunk 1 above)
Comment (guoyr) [reply evergreen.yml]: Can't argue with that!

**Pull request merged**